### DISCUSSION TITLE
How to use the configuration on a different machine?

### DISCUSSION THREAD

**Discussion by ukreddy-erwin** [abraunegg/onedrive#1615] [Q&A] [answered]
After installation, we need to login once and then can use the sync commands on a folder. Can I copy any configuration files from this machine so that I don't have to do this login process manually on a new machine? 

If it works then I can create an image for the machine and use it as part of GitHub actions.

**Comment by abraunegg** [comment, marked as answer]
@ukreddy-erwin 
> Can I copy any configuration files from this machine so that I don't have to do this login process manually on a new machine?

You can copy the 'refresh_token' & 'config' files between systems, with the following caveats:
1. You store the source files in a secure manner
2. You copy the files in a secure manner to the target systems
3. You constantly update|refresh the 'refresh_token' file - as this file will get updated during application use. If you do not update | keep the source 'refresh_token' file up-to-date, you will, quite quickly find that the client is asking you to re-auth as your copied token file is now invalid.

**Reply by ukreddy-erwin** [reply]
By source file you are referring to the files I upload or the configuration file related to onedrive client?
If it is the first case, I will keep only one way upload instead of syncing two way.
Or can you please elaborate the 3rd point if this is not the case.

**Reply by abraunegg** [reply]
@ukreddy-erwin 
Please can you re-read my response in regards to your question

**Reply by ukreddy-erwin** [reply]
So how frequent we need to update the refresh_token?  If it is within minutes, then it is not helpful by copying these files then

**Reply by abraunegg** [reply]
You generally need to update / refresh the token with the updated data at least daily.

**Reply by ukreddy-erwin** [reply]
ok, I will test it once and see how it goes
